libexec/scoop-update.ps1

```diff
@@ -57,12 +57,23 @@ if(($PSVersionTable.PSVersion.Major) -lt 5) {
 
 function update_scoop() {
     # check for git
-    if(!(Test-CommandAvailable git)) { abort "Scoop uses Git to update itself. Run 'scoop install git' and try again." }
+    if (!(Test-CommandAvailable git)) { abort "Scoop uses Git to update itself. Run 'scoop install git' and try again." }
+
+    $SCOOP_HOLD = get_config SCOOP_HOLD $false
+    $last_update = last_scoop_update
+    if ($null -eq $last_update) {
+        $last_update = [System.DateTime]::Now.ToString('o')
+        set_config lastUpdate $last_update | Out-Null
+    }
+    if (((New-TimeSpan ($last_update)).Days) -lt $SCOOP_HOLD) {
+        warn "Skipping self-update as 'SCOOP_HOLD' has been set to $([int]$SCOOP_HOLD) day$(if($SCOOP_HOLD -gt 1){'s'})..."
+        warn "If you want to update Scoop itself immediately, use 'scoop unhold scoop; scoop update'."
+        return
+    } else {
+        set_config SCOOP_HOLD $null | Out-Null
+    }
```

Comment:
```suggestion
    $last_update = last_scoop_update
    $now = [System.DateTime]::Now
    if ($null -eq $last_update) {
        $last_update = $now.ToString('o')
    }
    $update_until = get_config update_until
    if ($null -ne $update_until -and $update_until.GetType() -eq [System.String]) {
        try {
            $update_until = [System.DateTime]::Parse($update_until)
        } catch {
            $update_until = $null
        }
    }
    if ($null -ne $update_until -and $now -lt $update_until) {
        warn "Skipping self-update until $($update_until.ToLocalTime())..."
        warn "If you want to update Scoop itself immediately, use 'scoop unhold scoop; scoop update'."
        return
    } else {
        set_config update_until $null | Out-Null
    }
```

Reply:
There is no need for a new config value. Just letting the `lastUpdate` stay the same and adding `SCOOP_HOLD` to it returns the unhold datetime. :smile: